singularity-eos/eos/eos_mgusup.hpp

```diff
@@ -0,0 +1,365 @@
+  // Thermal Bulk Modulus added AEM Dec 2022
+  PORTABLE_INLINE_FUNCTION Real TBulkModulusFromDensityTemperature(
+      const Real rho, const Real temp, Real *lambda = nullptr) const;
+  PORTABLE_INLINE_FUNCTION Real BulkModulusFromDensityTemperature(
+      const Real rho, const Real temp, Real *lambda = nullptr) const;
+  PORTABLE_INLINE_FUNCTION Real BulkModulusFromDensityInternalEnergy(
+      const Real rho, const Real sie, Real *lambda = nullptr) const;
+  // Thermal expansion coefficient added AEM 2022
+  PORTABLE_INLINE_FUNCTION Real TExpansionCoeffFromDensityTemperature(
+      const Real rho, const Real temp, Real *lambda = nullptr) const;
+  PORTABLE_INLINE_FUNCTION Real GruneisenParamFromDensityTemperature(
+      const Real rho, const Real temp, Real *lambda = nullptr) const {
+    return robust::ratio(_G0 * _rho0, rho);
+  }
+  PORTABLE_INLINE_FUNCTION Real GruneisenParamFromDensityInternalEnergy(
+      const Real rho, const Real sie, Real *lambda = nullptr) const {
+    return robust::ratio(_G0 * _rho0, rho);
+  }
+  PORTABLE_INLINE_FUNCTION void FillEos(Real &rho, Real &temp, Real &energy, Real &press,
+                                        Real &cv, Real &bmod, const unsigned long output,
+                                        Real *lambda = nullptr) const;
+  PORTABLE_INLINE_FUNCTION
+  void ValuesAtReferenceState(Real &rho, Real &temp, Real &sie, Real &press, Real &cv,
+                              Real &bmod, Real &dpde, Real &dvdt,
+                              Real *lambda = nullptr) const;
+  // Generic functions provided by the base class. These contain e.g. the vector
+  // overloads that use the scalar versions declared here
+  SG_ADD_BASE_CLASS_USINGS(MGUsup)
+  PORTABLE_INLINE_FUNCTION
+  int nlambda() const noexcept { return 0; }
+  static constexpr unsigned long PreferredInput() { return _preferred_input; }
+  static inline unsigned long scratch_size(std::string method, unsigned int nelements) {
+    return 0;
+  }
+  static inline unsigned long max_scratch_size(unsigned int nelements) { return 0; }
+  PORTABLE_INLINE_FUNCTION void PrintParams() const {
+    static constexpr char st[]{"MGUsup Params: "};
+    printf("%s rho0:%e T0:%e Cs:%e s:%e\n  G0:%e Cv0:%e E0:%e S0:%e\n", st, _rho0, _T0,
+           _Cs, _s, _G0, _Cv0, _E0, _S0);
+    printf("\n\n");
+  }
+  // Density/Energy from P/T not unique, if used will give error
+  PORTABLE_INLINE_FUNCTION void
+  DensityEnergyFromPressureTemperature(const Real press, const Real temp, Real *lambda,
+                                       Real &rho, Real &sie) const;
+  inline void Finalize() {}
+  static std::string EosType() { return std::string("MGUsup"); }
+  static std::string EosPyType() { return EosType(); }
+
+ private:
+  static constexpr const unsigned long _preferred_input =
+      thermalqs::density | thermalqs::specific_internal_energy;
+  Real _rho0, _T0, _Cs, _s, _G0, _Cv0, _E0, _S0;
+  void CheckMGUsup();
```

Comment:
It's good practice to name private functions with a leading or trailing underscore. Can you add one to this name?

Reply:
OK. Sounds good.

Reply:
Fixed